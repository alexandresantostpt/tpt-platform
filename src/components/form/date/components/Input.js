import React from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

const StyledInput = styled.input`
    background-color: transparent;
    border: none;
    border-bottom: 1px solid #dcdcdc;
    color: ${colors.text};
    font-size: 1rem;
    font-weight: 300;
    padding: 5px;
    transition: all 0.15s linear;
    width: calc(100% - 0.625rem);

    &:focus {
        outline: none;
    }
`

const Input = props => <StyledInput readOnly type="text" {...props} />

export default Input
