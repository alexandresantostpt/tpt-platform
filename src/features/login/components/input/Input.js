import PropTypes from 'prop-types'
import React, { useState } from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

import { i18n } from '@i18n'

import If from '@components/if/If'

const StyledBox = styled.div`
    display: flex;
    font-size: 1.125rem;
    margin-bottom: 2rem;
    padding-top: 1.5rem;
    position: relative;
    width: 100%;

    *:focus {
        outline: none;
    }
`

const StyledButton = styled.button`
    background-color: white;
    border: none;
    bottom: 0.3125rem;
    color: ${colors.inputButtonGray};
    cursor: pointer;
    font-size: 75%;
    position: absolute;
    right: 0;
    text-transform: uppercase;
    width: auto;
`

const StyledLabel = styled.label`
    color: ${colors.inputGray};
    cursor: text;
    position: absolute;
    text-transform: uppercase;
    top: 2rem;
    transition: all 200ms;
`

const StyledInput = styled.input`
    border: 0;
    border-bottom: 0.0875rem solid ${colors.inputGray};
    color: ${colors.text};
    font-size: 1.125rem;
    outline: 0;
    padding: 0.3125rem ${({ type }) => (type === 'password' ? '5rem' : '0')} 0.3125rem 0;
    width: 100%;

    &:focus + ${StyledLabel} {
        font-size: 75%;
        transform: translate3d(0, -150%, 0);
    }

    &:valid + ${StyledLabel} {
        font-size: 75%;
        transform: translate3d(0, -150%, 0);
    }
`

const Input = ({ id, onChange, placeholder, type }) => {
    const [text, updateText] = useState('mostrar')
    const [typeUsed, updateTypeUsed] = useState(type)

    const top = () => {
        if (typeUsed === 'text') {
            updateTypeUsed('password')
            updateText(i18n.t('buttons.show'))
        } else {
            updateTypeUsed('text')
            updateText(i18n.t('buttons.hide'))
        }
    }

    return (
        <StyledBox>
            <StyledInput id={id} onChange={({ target }) => onChange(target.value)} required type={typeUsed} />
            <StyledLabel htmlFor={id}> {placeholder} </StyledLabel>
            <If condition={type === 'password'}>
                <StyledButton onClick={() => top()} type="button">
                    {text}
                </StyledButton>
            </If>
        </StyledBox>
    )
}

Input.defaultProps = {
    placeholder: ''
}

Input.propTypes = {
    id: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
    type: PropTypes.string.isRequired
}

export default Input
