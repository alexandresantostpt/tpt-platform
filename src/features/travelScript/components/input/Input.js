import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { colors } from '@helpers/colors'

const StyledInput = styled.input`
    background: ${colors.white};
    border: solid 0.5px ${colors.suvaGray};
    border-radius: 6px;
    box-sizing: border-box;
    color: ${colors.text};
    font-size: 15px;
    max-width: 350px;
    min-width: 236px;
    padding: 6px 19px;
    width: 40%;
    &:focus {
        outline: none;
    }

    &::-webkit-input-placeholder,
    &::-moz-placeholder,
    &:-ms-input-placeholder,
    &:-moz-placeholder {
        color: ${colors.zambezi};
    }
`

const Input = ({ onChange, placeholder, ...props }) => (
    <StyledInput onChange={onChange} placeholder={placeholder} type="text" {...props} />
)

Input.defaultProps = {
    placeholder: '',
    value: ''
}

Input.propTypes = {
    onChange: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string
}

export default Input
