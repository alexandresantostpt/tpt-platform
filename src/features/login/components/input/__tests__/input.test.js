import React from 'react'
import renderer from 'react-test-renderer'

import Input from '../Input'

describe('Tests for Input component', () => {
    it('Should rendering without crash', () => {
        const tree = renderer.create(<Input id="teste" onChange={jest.fn()} type="text" />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
