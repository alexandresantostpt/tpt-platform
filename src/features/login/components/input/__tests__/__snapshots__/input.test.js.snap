// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Input component Should rendering without crash 1`] = `
<div
  className="Input__StyledBox-sc-4bhgmp-0 gcStMN"
>
  <input
    className="Input__StyledInput-sc-4bhgmp-3 hoXjZP"
    id="teste"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <label
    className="Input__StyledLabel-sc-4bhgmp-2 lcosGj"
    htmlFor="teste"
  >
     
    
     
  </label>
</div>
`;
